mautrix_facebook/portal.py

```diff
@@ -496,14 +499,22 @@ def optional_send_lock(self, user_id: str) -> Union[asyncio.Lock, FakeLock]:
         if message.sticker:
             event_ids = [await self._handle_facebook_sticker(intent, message.sticker,
                                                              message.reply_to_id)]
-        elif len(message.attachments) > 0:
-            event_ids = await asyncio.gather(
-                *[self._handle_facebook_attachment(intent, attachment, message.reply_to_id)
-                  for attachment in message.attachments])
-            event_ids = [event_id for event_id in event_ids if event_id]
-        if not event_ids and message.text:
-            event_ids = [await self._handle_facebook_text(intent, message)]
         else:
+            if message.text:
+                event_ids = [await self._handle_facebook_text(intent, message)]
+
+            if len(message.attachments) > 0:
+                # Get URLs in the original message to prevent sending unnecessary ShareAttachments
+                urls = self._urlRegex.findall(message.text)
+                # not great... attachment.original_url adds a trailing slash in urls like https://example.com
+                urls.extend([url + '/' for url in urls if url[-1] != '/'])
```

Comment:
I really don't like this, but I couldn't find why there is a trailing slash added to original_url